external/fv3fit/fv3fit/emulation/models/microphysics.py

```diff
@@ -323,16 +323,14 @@ def _assoc_conservative_precipitation(
     )
 
     out = model(inputs)
-    out[
-        fields.surface_precipitation.output_name
-    ] = thermo.conservative_precipitation_zhao_carr(
+    precip_scalar = thermo.conservative_precipitation_zhao_carr(
         specific_humidity_before=inputs[fields.specific_humidity.input_name],
         specific_humidity_after=out[fields.specific_humidity.output_name],
         cloud_before=inputs[fields.cloud_water.input_name],
         cloud_after=out[fields.cloud_water.output_name],
         mass=thermo.layer_mass(inputs[fields.pressure_thickness.input_name]),
     )
-
+    out[fields.surface_precipitation.output_name] = precip_scalar[:, None]
```

Comment:
Yah...not sure where to make this most visible. I started adding some warning to `score_model`. But the problem is not just `score` since the keras loss functions also probably broadcast the data before reducing it again. It just wasn't an issue there because the batch size is smaller.

Maybe `train` could error out...

Reply:
I think an error on target / prediction shape mismatch seems reasonable.  Maybe a small prediction before `.fit` just to verify